src/utils/handlers/handleFormSubmit.ts

```diff
@@ -3,16 +3,26 @@ import { SubmitPromise } from "@saleor/hooks/useForm";
 async function handleFormSubmit<T>(
   data: T,
   onSubmit: (data: T) => SubmitPromise,
-  setChanged: (changed: boolean) => void
+  setChanged: (changed: boolean) => void,
+  setEnableExitPrompt?: (value: boolean) => void
 ): Promise<boolean> {
-  const errors = await onSubmit(data);
-  const ok = errors?.length === 0;
+  const result = onSubmit(data);
```

Comment:
Didn't we want to wrap this in try/catch blocks and use promise rejection as a mechanism for submit errors?

Reply:
Resolving due to this issue being addressed in the next pr